src/attr.ts

```diff
@@ -73,6 +73,12 @@ export function initializeAttrs(instance: HTMLElement, names?: Iterable<string>)
   }
 }
 
+function getAttrNames(instance: HTMLElement | typeof HTMLElement): string[] {
+  if (!instance || instance === HTMLElement) return []
+  const proto = Object.getPrototypeOf(instance)
+  return (attrs.get(proto) || []).concat(getAttrNames(proto))
+}
```

Comment:
I'd be interested in seeing this without recursion. I feel a while loop might be simpler to read here.

Reply:
Yup. That’s better for sure.